Makefile

```diff
@@ -4,52 +4,88 @@ else
 ENV_PATH_SEP := :
 endif
 
-
 THIS_MAKEFILE_PATH := $(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST))
 THIS_DIR := $(shell cd $(dir $(THIS_MAKEFILE_PATH));pwd)
 
-NPM ?= $(NODE) $(shell which npm)
-NPM_BIN = $(shell npm bin)
-
 RED=`tput setaf 1`
 GREEN=`tput setaf 2`
+CYAN=`tput setaf 6`
 RESET=`tput sgr0`
 
-START_APP := @$(NPM_BIN)/electron .
-ELECTRON_TEST := ELECTRON_PATH=$(NPM_BIN)/electron $(NPM_BIN)/electron-mocha
-CONFIG := $(THIS_DIR)/desktop/config.json
-DESKTOP_CONFIG := $(THIS_DIR)/desktop-config
-BUILDER := $(THIS_DIR)/builder.js
-BUILD_CONFIG := $(THIS_DIR)/resource/build-scripts/build-config-file.js
-BUILD_DIR := $(THIS_DIR)/build
-PACKAGE_DMG := $(THIS_DIR)/resource/build-scripts/package-dmg.js
-PACKAGE_WIN32 := @$(NPM_BIN)/electron-builder
-CERT_SPC := $(THIS_DIR)/resource/secrets/automattic-code.spc
-CERT_PVK := $(THIS_DIR)/resource/secrets/automattic-code.pvk
 CALYPSO_DIR := $(THIS_DIR)/calypso
-CALYPSO_JS := $(CALYPSO_DIR)/public/build.js
-CALYPSO_CHANGES_STD := `find "$(CALYPSO_DIR)" -newer "$(CALYPSO_JS)" \( -name "*.js" -o -name "*.jsx" -o -name "*.json" -o -name "*.scss" \) -type f -print -quit | grep -v .m. | wc -l`
-CALYPSO_BRANCH = $(shell git --git-dir ./calypso/.git branch | sed -n -e 's/^\* \(.*\)/\1/p')
-WEBPACK_BIN := $(NPM_BIN)/webpack
 
-# sets to 1 if NPM version is >= 3
-NPMGTE3 := $(shell expr `npm -v | cut -f1 -d.` \>= 3)
-MAKENSIS_VERSION := $(shell expr `makensis -VERSION 2>/dev/null`)
+CHECKMARK=✓
+
+CONFIG_ENV = 
+CALYPSO_ENV = desktop
+NODE_ENV = production
+BUILD_PLATFORM = mwl
+
+# Build sources
+# TODO: run tasks parallel when in dev mode
+build-source: checks build-config build-calypso build-desktop
+	@echo "$(CYAN)$(CHECKMARK) All parts built$(RESET)"
+
+# Start app
+start: build-source
+	@echo "$(CYAN)Starting app...$(RESET)"
+
+	@npx electron .
+
+# Build config
+build-config: 
+	@node $(THIS_DIR)/resource/build-scripts/build-config-file.js $(CONFIG_ENV)
+
+	@echo "$(CYAN)$(CHECKMARK) Config created$(RESET)"
+
+# BUild calypso bundle
+build-calypso: 
+	@cd $(CALYPSO_DIR) && NODE_ENV=$(NODE_ENV) CALYPSO_ENV=$(CALYPSO_ENV) npm run -s build
+
+	echo "$(CYAN)$(CHECKMARK) Calypso built$(RESET)"
+
+# Build desktop bundle
+build-desktop:
```

Comment:
where `make` can really speed things up (though not really for CircleCI) is when we add in dependency information. what does the build depend on? JS files? SCSS files? assets?

```Makefile
JS_FILES=$(shell find client -type f -name "*.js")
…

build-desktop: $(JS_FILES) $(SCSS_FILES) $(ASSETS)
```

obviously that list of files is incomplete but it's not too hard to make it more complete with some options to `find`. `make` will only rebuilt the app if files have changed.

further, should we end up only needing to update some files we can use the `$?` automatic variable which returns a space-separated list of files that have changed since the target was last built